tsl/src/compression/compression_dml.c

```diff
@@ -407,6 +422,8 @@ decompress_batches_scan(Relation in_rel, Relation out_rel, Relation index_rel, S
 	int num_filtered_rows = 0;
 	TM_Result result;
 	DecompressBatchScanDesc scan = NULL;
+	BatchMatcher *batch_matcher =
+		constraints && constraints->vectorized_filtering ? batch_matches_vectorized : batch_matches;
```

Comment:
Is it complicated to do both at the same time? DecompressChunk does this, some filters can be vectorized and some not, and both are supported at the same time.

Reply:
Ah its a bit more complicated, I'm gonna leave it out this first implementation to keep things simple. We can implement it as a further optimization down the line.